tests/func/test_repro.py

```diff
@@ -1420,3 +1420,45 @@ def _format_dvc_line(line):
 
 def _hide_md5(text):
     return re.sub(r"\b[a-f0-9]{32}\b", "<md5>", text)
+
+
+class TestReproDownstream(TestDvc):
+    def test(self):
+        # The dependency graph should look like this:
+        #
+        #       E
+        #      / \
+        #     D   F
+        #    / \   \
+        #   B   C   G
+        #    \ /
+        #     A
+        #
+        assert main(["run", "-o", "A", "echo A>A"]) == 0
+        assert main(["run", "-d", "A", "-o", "B", "echo B>B"]) == 0
+        assert main(["run", "-d", "A", "-o", "C", "echo C>C"]) == 0
+        assert main(["run", "-d", "B", "-d", "C", "-o", "D", "echo D>D"]) == 0
+        assert main(["run", "-o", "G", "echo G>G"]) == 0
+        assert main(["run", "-d", "G", "-o", "F", "echo F>F"]) == 0
+        assert main(["run", "-d", "D", "-d", "F", "-o", "E", "echo E>E"]) == 0
+
+        self._caplog.clear()
+
+        # We want the evaluation to move from B to E
+        #
+        #       E
+        #      /
+        #     D
+        #    /
+        #   B
+        #
+        assert main(["repro", "--downstream", "B.dvc"]) == 0
```

Comment:
`self.dvc.reproduce("B.dvc", downstream=True)` would return you a list of stages that it reproduced, so you don't have to rely on fragile parsing of logs.

Reply:
I was wondering if it was useful to have a way to test this on a CLI level -- an easy way to tell if a command was reproduced or not. What do you think, @efiop ? Should we rely on the module instead (`self.dvc.reproduce`)?

Reply:
@mroutis It makes sense to test that `--downstream` is properly passed from CLI to API. You could add a simple unit test for that separately.

Reply:
Also, @efiop , `self.dvc.reproduce` returns a list of stages that were _changed_ while recomputing the pipeline instead of returning the _evaluated_ stages.

Reply:
In this case, relying on parsing logs is a really fragile thing and using API is much better.

Reply:
@mroutis Just use `force=True` then :)